project/SparkBuild.scala

```diff
@@ -748,6 +739,13 @@ object Java8TestSettings {
 object TestSettings {
   import BuildCommons._
 
+  private val scalaBinaryVersion =
+    if (System.getProperty("scala-2.10") == "true") {
```

Comment:
Meh, this is fine for now; I can clean it up later if necessary as part of 2.12.